### PR TITLE
Electron first stab

### PR DESCRIPTION
**Pull request by notlaforge**
First crack at #137

Still a long ways to go:

- [ ] Packaging
- [ ] Dev builds tracking master
- [ ] Signing?
- [ ] Mac OS X icons
- [ ] 3rd party notice
- [ ] AAD login
- [ ] Automatic login on domain joined machine
- [ ] Offline works (with Emulator?)

### CODE REVIEW

**Comment by notlaforge** [comment]
Work in progress...

image

image

File: src/Common/CosmosClient.ts [hunk 1]
```diff
@@ -85,8 +85,7 @@ export function client(): Cosmos.CosmosClient {
     userAgentSuffix: "Azure Portal"
   };
 
-  // In development we proxy requests to the backend via webpack. This is removed in production bundles.
```
Comment (notlaforge) [inline src/Common/CosmosClient.ts:88]: I'm a bit concerned about this. What is the possible negative side effect of this if-statement not being removed in production bundles?

File: src/Desktop/app.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+function createMainWindow() {
+  if (main) {
+    return;
+  }
+
+  // Create the browser window.
+  main = new BrowserWindow({
+    icon: resolve(__dirname, "..", "..", "src", "Desktop", "cosmosdb.ico"),
+    width: 1024,
+    height: 800,
```
Comment (languy) [inline src/Desktop/app.ts:24]: I know it's WIP, but we probably want to define this as constant so we can make it configurable in the future?

File: src/Explorer/Tabs/MongoShellTab.ts [hunk 3]
```diff
@@ -31,8 +32,7 @@ export default class MongoShellTab extends TabsBase {
       const accountName = account && account.name;
       const mongoEndpoint = account && (account.properties.mongoEndpoint || account.properties.documentEndpoint);
 
-      this._runtimeEndpoint =
-        window.dataExplorerPlatform == PlatformType.Hosted ? AuthHeadersUtil.extensionEndpoint : "";
+      this._runtimeEndpoint = window.dataExplorerPlatform == PlatformType.Hosted ? configContext.BACKEND_ENDPOINT : "";
```
Comment (languy) [inline src/Explorer/Tabs/MongoShellTab.ts:35]: `===`

**Review by languy — APPROVED** [review]
Can't wait to start playing with it! :)

**Comment by southpolesteve** [comment]
@notlaforge I cross the same issue solved in this PR earlier in the week. I took this a step further and removed all usage of extensionEndpoint(). Now configContext.BACKEND_ENDPOINT is the authoritative source.

**Review by southpolesteve — APPROVED** [review]

**Pull request merged**